executionlayer/src/main/kotlin/maru/executionlayer/manager/ForkScheduleAwareExecutionLayerManager.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.executionlayer.manager
+
+import kotlinx.datetime.Clock
+import maru.config.consensus.ElFork
+import maru.config.consensus.qbft.QbftConsensusConfig
+import maru.consensus.ForksSchedule
+import maru.core.ExecutionPayload
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+
+class ForkScheduleAwareExecutionLayerManager(
+  val forksSchedule: ForksSchedule,
+  val executionLayerManagerMap: Map<ElFork, ExecutionLayerManager>,
+  val clock: Clock = Clock.System,
+) : ExecutionLayerManager {
+  init {
+    ElFork.entries.forEach {
+      require(executionLayerManagerMap.containsKey(it)) { "No execution layer manager provided for $it" }
+    }
+  }
+
+  internal fun getCurrentElFork(): ElFork {
+    val currentTimestamp = clock.now().epochSeconds
```

Comment:
I realized that this logic is somewhat working, but isn't good enough. One of the side effects of the ElSyncService is that it would start the El sync before the CL sync is complete. And if it sets the FCU to the past block heights, it needs to get the forks using the timestamps from the current state of the `BeaconChain`. This logic will start to work only when `BeaconChain` reaches the current fork

Reply:
> I do not understand the need to use the timestamp from the current state to decide on the EL Fork, the EL Node would switch independently based on the its clock and not the timestamp

Did you try it? Different newPayload methods need to be used depending on the EL node's current fork

Reply:
But we want `SYNCING` response